### ISSUE TITLE
plantuml-server service failing to start

### ISSUE THREAD

**Issue by dcarosone** [NixOS/nixpkgs#262893]
### Describe the bug

plantuml server fails to start; logs indicate usage error in the command being invoked 

### Steps To Reproduce

```nix
services.plantuml-server.enable = true;
```

Update unstable since.. roughly early last week.  

### Expected behavior
service starts and works as previously

### Screenshots

journalctl -u plantuml-server.service is an endless loop of retries like this (leading times and hostname trimmed)

```console
plantuml-server-start[2295851]: ERROR : Unknown module=[jsp] List available with --list-modules
plantuml-server-start[2295851]: Usage: java -jar $JETTY_HOME/start.jar [options] [properties] [configs]
plantuml-server-start[2295851]:        java -jar $JETTY_HOME/start.jar --help  # for more information
systemd[1]: plantuml-server.service: Main process exited, code=exited, status=247/n/a
systemd[1]: plantuml-server.service: Failed with result 'exit-code'.
systemd[1]: plantuml-server.service: Consumed 1.915s CPU time, no IP traffic.
```

### Additional context

I did notice a jetty upgrade recently, but I thought it was earlier than this - may or may not be related.

### Notify maintainers

@truh 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"

 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.58, NixOS, 23.11 (Tapir), 23.11.20231019.7c9cc5a`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.17.1`
 - channels(root): `""`
 - nixpkgs: `/nix/store/4jfc6vrkmq7z5pb651jh5b4kra5f1kwp-source`
```

**Comment by truh** [comment]
I've asked the developers for input. If anyone here has ideas how to make this work or whether we should downgrade to Jetty 11, I'm happy for inputs.

I was able to reproduce the problem with these commands:

```
nix build github:NixOS/nixpkgs/nixos-unstable#jdk -o jdk
nix build github:NixOS/nixpkgs/nixos-unstable#jetty -o jetty
nix build github:NixOS/nixpkgs/nixos-unstable#plantuml-server -o plantuml-server
jdk/bin/java -jar jetty/start.jar --module=deploy,http,jsp jetty.home=$PWD/jetty jetty.base=$PWD/plantuml-server
```

Using Jetty 11 from NixOS 23.05 it works as expect (web app can be accessed on localhost:8080/plantuml)

```
nix build github:NixOS/nixpkgs/nixos-23.05#jetty -o jetty
```

**Comment by dcarosone** [comment]
Will be interesting to see what they come back with, but a quick glance over the migration guide
https://eclipse.dev/jetty/documentation/jetty-12/programming-guide/index.html#pg-migration-11-to-12
and especially
https://eclipse.dev/jetty/documentation/jetty-12/programming-guide/index.html#pg-migration-11-to-12-api-changes

looks like a bunch of breaking changes (though that term isn't explicitly used that I see).

My guess is that unless they come back with a new release to adapt to these changes, we need to look at making it depend on 11, and perhaps allowing concurrent different versions of jetty as separate packages?

**Comment by emmanuelrosa** [comment]
I'm going to work on a Jetty 11 package. I've got a mess to untangle.

In the mean time, you can use a pinned version of Nixpkgs to pull in Jetty 11.

**Comment by dcarosone** [comment]
Has been resolved via deae39af
